src/machine_charm.py

```diff
@@ -138,17 +131,56 @@ def restart(self) -> None:
         """Restart grafana agent."""
         subprocess.run(["sudo", "snap", "restart", self._service])
 
-    @property
-    def is_machine(self) -> bool:
-        """Check if this is a machine charm."""
-        return True
-
     @property
     def _is_installed(self) -> bool:
         """Check if the Grafana Agent snap is installed."""
         package_check = subprocess.run("snap list | grep grafana-agent", shell=True)
         return True if package_check.returncode == 0 else False
 
+    @property
+    def _additional_integrations(self) -> Dict[str, Any]:
+        """Additional integrations for machine charms."""
+        node_exporter_job_name = (
+            f"juju_{self.model.name}_{self.model.uuid}_{self.model.app.name}_node-exporter"
+        )
+        return {
+            "node_exporter": {
+                "enabled": True,
+                "relabel_configs": [
+                    # Align the "job" name with those of prometheus_scrape
+                    {
+                        "target_label": "job",
+                        "regex": "(.*)",
+                        "replacement": node_exporter_job_name,
+                    },
+                ]
+                + self._principal_relabeling_config,
+            }
+        }
+
+    @property
+    def _additional_log_configs(self) -> List[Dict[str, Any]]:
+        """Additional logging configuration for machine charms."""
+        return [
+            {
+                "name": "log_file_scraper",
+                "clients": self._loki_consumer.loki_endpoints,
+                "scrape_configs": [
+                    {
+                        "job_name": "varlog",
+                        "static_configs": {
+                            "labels": {
+                                "__path__": "/var/log/*",
```

Comment:
we should probably change this to avoid trying to tail zip files and rotated logs.

```suggestion
                                "__path__": "/var/log/*log",
```